### PR TITLE
Branch ped bugfix

### LINKED ISSUE #167: [PE-D][Tester B] Bug for date validation in consultation
Steps to reproduce
1. createconsult d/30/02/2023 tt/23:59 n/Alex Yeoh

Expected
Error Message

Actual
Nothing happens, no error message shown

30/2/2023 is an invalid date which I expect an error message but there was no error message.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: Carlintyj/ped#1

### LINKED ISSUE #190: [PE-D][Tester C] Accepted Invalid Date
I am able to add task on invalid dates such as 29/30/31 Feb (Non leap year), 30/31 Feb(Leap year)

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: Jonyxzx/ped#5

### PR DESCRIPTION
**Pull request by Respirayson**

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/ViewTasksCommandParserTest.java [hunk 1]
```diff
@@ -52,14 +52,14 @@ public void parse_validTaskDescArgs_returnsViewTasksCommand() {
...
-        assertParseSuccess(parser, " " + PREFIX_TASK_DESCRIPTION + "prs \n \t book", expectedCommand);
+        assertParseSuccess(parser, " " + PREFIX_TASK_DESCRIPTION + "prs \t book", expectedCommand);
     }
 
     @Test
     public void parse_validTaskProgressArgs_returnsViewTasksCommand() {
         // no leading and trailing whitespaces
         ViewTasksCommand expectedCommand =
-                new ViewTasksCommand(new TaskProgressContainsKeywordsPredicate(Arrays.asList("done", "pending")));
+                new ViewTasksCommand(new TaskProgressContainsKeywordsPredicate(Arrays.asList("DONE", "PENDING")));
```
Comment (raydenlim) [inline src/test/java/seedu/address/logic/parser/ViewTasksCommandParserTest.java:62]: This suppose to test for case sensitivity?

Comment (raydenlim) [reply src/test/java/seedu/address/logic/parser/ViewTasksCommandParserTest.java:62]: same for line 73

File: src/main/java/seedu/address/logic/parser/ViewTasksCommandParser.java [hunk 2]
```diff
@@ -46,27 +49,42 @@ public ViewTasksCommand parse(String args) throws ParseException {
...
+                checkedTaskProgressKeywords.add(ParserUtil.parseTaskProgress(str).name());
+            }
+
+            predicate = new TaskProgressContainsKeywordsPredicate(checkedTaskProgressKeywords);
         }
 
         if (argMultimap.getValue(PREFIX_DATE).isPresent()) {
-            String[] dateKeywords = argMultimap.getValue(PREFIX_DATE).get().split("\\s+");
+            String[] dateKeywords = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).get())
+                    .format(DateTimeFormatter.ofPattern("dd/MM/yyyy")).split("\\s+");
```
Comment (raydenlim) [inline src/main/java/seedu/address/logic/parser/ViewTasksCommandParser.java]: not sure if we should use yyyy or uuuu here.
uuuu might be safer

**Review by raydenlim — APPROVED** [review]
LGTM!

File: src/test/java/seedu/address/logic/parser/ViewTasksCommandParserTest.java (hunk 1 above)
Comment (Respirayson) [reply src/test/java/seedu/address/logic/parser/ViewTasksCommandParserTest.java:62]: the enum.name() returns the capital form so the equality check will fail if its not capitalised

**Pull request merged**